pyscf/cc/ccsd.py

```diff
@@ -746,6 +759,8 @@ class CCSD(lib.StreamObject):
     diis_start_energy_diff = getattr(__config__, 'cc_ccsd_CCSD_diis_start_energy_diff', 1e9)
 
     direct = getattr(__config__, 'cc_ccsd_CCSD_direct', False)
+    cc_async = getattr(__config__, 'cc_ccsd_CCSD_direct', True)
```

Comment:
I'm not to keen on the name 'async' as it seems to clash with 

https://docs.python.org/3/library/asyncio-task.html

Maybe some other name? However, how would you like it defined in the global config? In lib/misc.py you've defined 'ASYNC_IO', but I would much rather prefer something that you can switch on/off for the individual methods... so I can implement ci_cisd_CISD_async as well in exactly the same way as for cc_ccsd_CCSD_async, but then we still end up with ci_async and cc_async (which is fine IMO, but you should have the last say).

Reply:
I agree cisd should have its config flag rather than using the one from ccsd. I think using the same name for this attribute looks more consistent in the whole package. I don't know async is a built-in keyword in python 3. Of cause, another name is better here. What about "async_io"?

Reply:
OK, almost all CI and CC tests now run through (sorry, should of course have tested this beforehand), except these (which I believe are unrelated to my changes):

======================================================================
ERROR: test_ucc_update_amps (__main__.KnownValues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_eom_uccsd.py", line 180, in test_ucc_update_amps
    self.assertAlmostEqual(abs(r1g-gcc1.spatial2spin(t1, orbspin)).max(), 0, 6)
  File "/Users/janusje/anaconda3/lib/python3.5/unittest/case.py", line 870, in assertAlmostEqual
    if round(abs(second-first), places) == 0:
TypeError: type NPArrayWithTag doesn't define __round__ method

======================================================================
FAIL: test_ERIS (__main__.KnownValues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_gccsd.py", line 58, in test_ERIS
    self.assertAlmostEqual(lib.finger(eris.oooo), -18.784809755855356, 9)
AssertionError: 1.4675375917253746 != -18.784809755855356 within 9 places

======================================================================
ERROR: test_update_lambda_complex (__main__.KnownValues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_rccsd_lambda.py", line 143, in test_update_lambda_complex
    self.assertAlmostEqual(abs(l1new[0::2,0::2]-l1new_ref).max(), 0, 9)
  File "/Users/janusje/anaconda3/lib/python3.5/unittest/case.py", line 870, in assertAlmostEqual
    if round(abs(second-first), places) == 0:
TypeError: type NPArrayWithTag doesn't define __round__ method

======================================================================
ERROR: test_update_lambda_real (__main__.KnownValues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_uccsd_lambda.py", line 77, in test_update_lambda_real
    self.assertAlmostEqual(abs(addons.spatial2spin(l1, orbspin)-l1ref).max(), 0, 8)
  File "/Users/janusje/anaconda3/lib/python3.5/unittest/case.py", line 870, in assertAlmostEqual
    if round(abs(second-first), places) == 0:
TypeError: type NPArrayWithTag doesn't define __round__ method